scipy/optimize/_root.py

```diff
@@ -637,3 +639,169 @@ def _root_krylov_doc():
             See `scipy.sparse.linalg.lgmres` for details.
     """
     pass
+
+
+_iter = 100
+_xtol = 1e-12
+_rtol = np.finfo(float).eps * 2
+
+def root_scalar(f, a, b, args=(), method='brentq', xtol=_xtol, rtol=_rtol,
+                maxiter=_iter, disp=True):
+    """
+    Find a root of a function in a sign-changing interval.
+
+    Parameters
+    ----------
+    f : function
+        Python function returning a number. The function must be
+        continuous and `f(a)` and `f(b)` must have opposite signs.
+
+    a : number
+        One end of the bracketing interval `[a,b]`.
+    b : number
+        The other end of the bracketing interval `[a,b]`.
+    args : tuple, optional
+        Extra arguments for the function `f`. In the code `f` is called by
+        ``apply(f, (x)+args)``.
+    method : str, optional
+        Type of solver. Should be one of
+
+            - ``'brentq'`` :ref:`(see here) <root-scalar-brentq>`
+            - ``'brenth'`` :ref:`(see here) <root-scalar-brenth>`
+            - ``'ridder'`` :ref:`(see here) <root-scalar-ridder>`
+            - ``'bisect'`` :ref:`(see here) <root-scalar-bisect>`
+
+    xtol : number, optional
+        The routine converges when a root is known to lie within `xtol` of the
+        value return. Should be >= 0.  The routine modifies this to take into
+        account the relative precision of doubles.
+    rtol : number, optional
+        The routine converges when a root is known to lie within `rtol` times
+        the value returned of the value returned. Should be >= 0. Defaults to
+        ``np.finfo(float).eps * 2``.
+    maxiter : number, optional
+        If convergence is not achieved in `maxiter` iterations, an error is
+        raised.  Must be >= 0.
+    disp : bool, optional
+        If ``True``, raise a RuntimeError if the algorithm didn't converge.
```

Comment:
`disp` parameter is usually used to control the verbosity. Not sure this is a good idea to raise an exception in case of non-convergence, the final result should be used to indicate that I think.